From 6f4647c3908900d5ed37635ab28873c038ba3091 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:50:05 -0400
Subject: [ethernet] bnx2x: Fix static checker warning regarding `txdata_ptr'

Message-id: <1409946615-5862-61-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91601
O-Subject: [RHEL7.1 PATCH 60/70] bnx2x: Fix static checker warning regarding `txdata_ptr'
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 1fc3de94076cd0c1e53e5ba8061763b33c7b27c2
Author: Yuval Mintz <Yuval.Mintz@qlogic.com>
Date:   Tue Aug 26 10:24:41 2014 +0300

    bnx2x: Fix static checker warning regarding `txdata_ptr'

    Incorrect checking of array instead of array contents in panic_dump
    flow - results of commit e261199872a2 ("bnx2x: Safe bnx2x_panic_dump()").

    Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Yuval Mintz <Yuval.Mintz@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
index 1d13f8d..2b3b5f2 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
@@ -1005,7 +1005,7 @@ void bnx2x_panic_dump(struct bnx2x *bp, bool disable_int)
   /* Tx */
   for_each_cos_in_tx_queue(fp, cos)
   {
-   if (!fp->txdata_ptr)
+   if (!fp->txdata_ptr[cos])
     break;
 
    txdata = *fp->txdata_ptr[cos];
@@ -1160,7 +1160,7 @@ void bnx2x_panic_dump(struct bnx2x *bp, bool disable_int)
   for_each_cos_in_tx_queue(fp, cos) {
    struct bnx2x_fp_txdata *txdata = fp->txdata_ptr[cos];
 
-   if (!fp->txdata_ptr)
+   if (!fp->txdata_ptr[cos])
     break;
 
    if (!txdata.tx_cons_sb)
-- 
1.7.1